lms/djangoapps/courseware/migrations/0014_auto__add_studentmodulehistoryextended.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import dbs
+from south.v2 import SchemaMigration
+from django.db import models
+
+default_db = dbs['default']
+csmh_db = dbs['student_module_history']
+
+class Migration(SchemaMigration):
+
+    def forwards(self, orm):
+
+        # Archive 'StudentModuleHistory'
+        default_db.rename_table('courseware_studentmodulehistory', 'courseware_studentmodulehistoryarchive')
+
+        # Create a new 'StudentModuleHistory' with an extended key space
+        csmh_db.create_table('courseware_studentmodulehistory', (
+            ('id', self.gf('courseware.fields.UnsignedBigIntAutoField')(primary_key=True)),
+            ('course_key', self.gf('xmodule_django.models.CourseKeyField')(max_length=255)),
+            ('usage_key', self.gf('xmodule_django.models.UsageKeyField')(max_length=255)),
+            ('username', self.gf('django.db.models.fields.CharField')(max_length=255)),
+            ('version', self.gf('django.db.models.fields.CharField')(db_index=True, max_length=255, null=True, blank=True)),
+            ('created', self.gf('django.db.models.fields.DateTimeField')(db_index=True)),
+            ('state', self.gf('django.db.models.fields.TextField')(null=True, blank=True)),
+            ('grade', self.gf('django.db.models.fields.FloatField')(null=True, blank=True)),
+            ('max_grade', self.gf('django.db.models.fields.FloatField')(null=True, blank=True)),
+        ))
+
+        if not default_db.dry_run:
+            last = orm.StudentModuleHistoryArchive.objects.all().latest('id').id
+
+            initial_id = last + 10000
```

Comment:
Yes, I'd like @e0d to help inform what that number should be, based on our current data use patterns.